Source/Core/TerrainMesh.js

```diff
@@ -104,7 +127,252 @@ define([
+        var clipRectangleUMax = (clipRectangle.east - thisRectangle.west) / (thisRectangle.east - thisRectangle.west);
+        var clipRectangleVMin = (clipRectangle.south - thisRectangle.south) / (thisRectangle.north - thisRectangle.south);
+        var clipRectangleVMax = (clipRectangle.north - thisRectangle.south) / (thisRectangle.north - thisRectangle.south);
+
+        var indices;
+        var first;
+        var second;
+        var edgeCoordinate;
+        var compareU;
+
+        switch (tileEdge) {
+            case TileEdge.WEST:
+                indices = this.westIndicesSouthToNorth;
+                first = 0.0;
+                second = 1.0;
+                edgeCoordinate = transformTextureCoordinate(clipRectangleUMin, clipRectangleUMax, 0.0);
+                compareU = false;
+                break;
+            case TileEdge.NORTH:
+                indices = this.northIndicesWestToEast;
+                first = 0.0;
+                second = 1.0;
+                edgeCoordinate = transformTextureCoordinate(clipRectangleVMin, clipRectangleVMax, 1.0);
+                compareU = true;
+                break;
+            case TileEdge.EAST:
+                indices = this.eastIndicesNorthToSouth;
+                first = 1.0;
+                second = 0.0;
+                edgeCoordinate = transformTextureCoordinate(clipRectangleUMin, clipRectangleUMax, 1.0);
+                compareU = false;
+                break;
+            case TileEdge.SOUTH:
+                indices = this.southIndicesEastToWest;
+                first = 1.0;
+                second = 0.0;
+                edgeCoordinate = transformTextureCoordinate(clipRectangleVMin, clipRectangleVMax, 0.0);
+                compareU = true;
+                break;
+        }
+
+        var encoding = this.encoding;
+        var vertices = this.vertices;
+        var southMercatorY;
+        var oneOverMercatorHeight;
+        var lastUOrV;
+
+        var destinationStride = 6;
+        if (encoding.hasVertexNormals) {
+            destinationStride += 2;
+        }
+        if (encoding.hasWebMercatorT) {
+            ++destinationStride;
+        }
+
+        if (encoding.hasWebMercatorT) {
+            southMercatorY = WebMercatorProjection.geodeticLatitudeToMercatorAngle(clipRectangle.south);
+            oneOverMercatorHeight = 1.0 / (WebMercatorProjection.geodeticLatitudeToMercatorAngle(clipRectangle.north) - southMercatorY);
+        }
+
+        for (var i = 0; i < indices.length; ++i) {
+            var index = indices[i];
+
+            var uv = encoding.decodeTextureCoordinates(vertices, index, uvScratch);
+            var u = transformTextureCoordinate(clipRectangleUMin, clipRectangleUMax, uv.x);
+            var v = transformTextureCoordinate(clipRectangleVMin, clipRectangleVMax, uv.y);
+            var uOrV = compareU ? u : v;
+
+            var inside = uOrV >= 0.0 && uOrV <= 1.0;
```

Comment:
Just style, but here and throughout, please wrap the right hand side of these expressions in parentheses.